### ISSUE TITLE
Should not chmod 777 /var/tezos 

### ISSUE THREAD

**Issue by harryttd** [oxheadalpha/tezos-k8s#378]
In config-generator.sh we run:
https://github.com/oxheadalpha/tezos-k8s/blob/4f3488153c5b2f8990552c45eab46e832b33c061/utils/config-generator.sh#L8-L9

The node on its initial creation will create files/dirs with default perms set by the Tezos docker image:
```shell
~ $ ls -l /var/tezos/ /var/tezos/node /var/tezos/node/data/
/var/tezos/:
total 24
drwxrwxrwx    2 root     users         4096 Feb 23 15:18 client
drwxrwxrwx    2 root     users        16384 Feb 23 15:18 lost+found
drwxr-xr-x    3 tezos    nogroup       4096 Feb 23 15:18 node

/var/tezos/node:
total 4
drwx------    4 tezos    nogroup       4096 Feb 23 15:19 data

/var/tezos/node/data/:
total 20
drwxr-xr-x    3 tezos    nogroup       4096 Feb 23 15:19 context
-rw-r--r--    1 tezos    nogroup        300 Feb 23 15:19 identity.json
-rw-r--r--    1 tezos    nogroup          1 Feb 23 15:18 lock
drwxr-xr-x    4 tezos    nogroup       4096 Feb 23 15:19 store
-rw-r--r--    1 tezos    nogroup         22 Feb 23 15:18 version.json
~ $
```

Only the `/var/tezos/client` is `777` because we create it in the script and then immediately run `chmod 777`. The node data dir doesn't exist yet and hence the less privileged perms.

If the node never restarts, these permissions will remain. If the pod restarts, now that the data dir does exist the `chmod 777` will give unnecessary perms. It will also change the group from `nogroup` to `users`:
```shell
 ~ $ ls -l /var/tezos/ /var/tezos/node /var/tezos/node/data/
/var/tezos/:
total 24
drwxrwxrwx    2 root     users         4096 Feb  1 02:41 client
drwxrwxrwx    2 root     users        16384 Feb  1 02:41 lost+found
drwxrwxrwx    3 tezos    users         4096 Feb  1 02:41 node

/var/tezos/node:
total 4
drwxrwxrwx    4 tezos    users         4096 Feb 18 19:58 data

/var/tezos/node/data/:
total 1508
drwxrwxrwx    3 tezos    users         4096 Feb  1 02:41 context
-rwxrwxrwx    1 tezos    users          300 Feb  1 02:41 identity.json
-rwxrwxrwx    1 tezos    users            1 Feb 18 20:07 lock
-rwxrwxrwx    1 tezos    users      1522573 Feb 18 19:58 peers.json
drwxrwxrwx    4 tezos    users         4096 Feb  1 02:41 store
-rwxrwxrwx    1 tezos    users           22 Feb  1 02:41 version.json
```

There should not be this inconsistency of the file perms. They aren't necessary and are a potential security risk. I also don't believe that `/var/tezos/client` should need `777` perms.

**Comment by harryttd** [comment]
We should also confirm `mkdir -p` in the snapshot downloader https://github.com/oxheadalpha/tezos-k8s/blob/8c0b449ce3/utils/snapshot-downloader.sh#L55 doesn't give more perms than what the tezos node does give by default.

Essentially, any time we create files/dirs that the tezos node would create by default, we should give the exact same perms.

**Comment by harryttd** [comment]
closed via #547
